### PR TITLE
Bluetooth: Mesh: Distributor sample: Enable shell over RTT

### PR DESCRIPTION
**Pull request by PavelVPV**
This PR enables shell commands over RTT backend and allows to run
mcumgr over UART allowing to interact with the sample over mcumgr
client without a need in a 3rd device.

### CODE REVIEW

**Comment by alxelax** [comment]
Hm... what to do customers who use shell on their own boards? Seems it is possible to use shell only if you have built-in segger jtag. Shouldn't we just add doc description for use case that you try to solve?

File: samples/bluetooth/mesh/dfu/distributor/README.rst [hunk 1]
```diff
@@ -167,7 +167,7 @@ A firmware image can be uploaded to the device in two ways:
 
 For out-of-band upload, the sample uses the image management subsystem of the :ref:`zephyr:mcu_mgr`.
 The management subsystem uses the Simple Management Protocol (SMP), provided by the Mcumgr library, to exchange commands and data between the SMP server (the sample device) and the SMP client.
-This sample supports Bluetooth Low Energy as the SMP transport.
+This sample supports Bluetooth Low Energy and UART as the SMP transport.
```
Comment (hieuv) [inline samples/bluetooth/mesh/dfu/distributor/README.rst:170]: Do you mean RTT here?

Comment (PavelVPV) [reply samples/bluetooth/mesh/dfu/distributor/README.rst:170]: No. This is for mcumgr: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/services/device_mgmt/mcumgr.html#configuring-the-transport. BLE transport allows to use nRF Connect Device Manager to interact with the device, or mcumgr client but with bluetooth controller running on host which requires a 3rd device, e.g. nRF52 with hci_uart sample (https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/samples/bluetooth/hci_uart/README.html#using-the-controller-with-qemu-and-native-posix). UART transport allows to run mcumgr client and interact with the device over shell without 3rd-party component. Enabling RTT shell backend allows to execute shell commands on the sample using e.g. RTT Viewer. You can still execute shell commands over UART but using mcumgr.

Comment (hieuv) [reply samples/bluetooth/mesh/dfu/distributor/README.rst:170]: Ah sorry I mixed things up. You are right.

**Review by ludvigsj — APPROVED** [review]

**Review by Andrewpini — APPROVED** [review]

**Review by mia-ko — COMMENTED** [review]
Do we need a changelog entry for this?

**Comment by PavelVPV** [comment]
> Do we need a changelog entry for this?

Right! Added.

**Review by mia-ko — APPROVED** [review]

**Review by hieuv — APPROVED** [review]

**Pull request merged**